backend/settings.py

```diff
@@ -236,6 +234,14 @@
     # Enables Proxies that set headers
     USE_X_FORWARDED_HOST = os.getenv('USE_X_FORWARDED_HOST', True)
 
+# Set CSP_FRAME_SRC to the yours Canvas domains
+CSP_FRAME_ANCESTORS = ["'self'",] + os.getenv('CSP_FRAME_ANCESTORS', '').split(',')
+# Allow inline scripts and Google assets. I don't think these need to be configurable
+CSP_SCRIPT_SRC = ["'self'", "'unsafe-inline'", "'unsafe-eval'", "www.google-analytics.com"]
```

Comment:
Is `unsafe-eval` necessary? Seems to work for me.

I don't think we have `unsafe-inline` set here for CCM, but it seems to be required. Do you know if this is because of how `pylti1p3` works? Or some other reason specific to this tool?

Reply:
Yeah, it looks necessary for the pyltip3. It has inline scripts. I don't know if it's necessary for anything else. It might be possible to improve this with hashes or even better suggesting to the author to add a nonce to the script tags. Then the hashes wouldn't change if the code changes.

I've [never done this](https://content-security-policy.com/hash/) but can look into seeing how much work it would be. 

I'm not sure if `unsafe-eval` is needed.

Reply:
I've removed `unsafe-eval.` 

These scripts are in PyLTI, and I can't add them as a SHA because the script actually changes each time. (I noticed this as I kept on adding new SHA's on each reload and eventually gave up) 

It looks like you can use this [CSP library](https://django-csp.readthedocs.io/en/latest/nonce.html) to generate nonce for scripts and include them, but we'd have to also include this as a dependency in the PyLTI library for this to work or create some of our own classes or middleware? It looks like there's been some [desire to get this into Django Core](https://code.djangoproject.com/ticket/15727) but that's almost 11 years old. 

I'm not sure how else to deal with this the "best" way. The easiest is just to leave `unsafe-inline` there.

Reply:
Not sure of the implications for `unsafe-inline` for scripts, but I'll trust your CSP expertise here. Penetration testing folks would also tell us if it's an issue.

Reply:
Yeah, we learned a little more from CCM and this but each app may have some differences in their security policy. We could simplify MyLA's though as very few of these options are ever necessary to be configured to setup the app.

Ideally we wouldn't have to allow unsafe-inline but we'd have to do more work around the packages we use to be able to remove that option. And it might need to be something supported by the framework like Django. 

I think since CSP passes that nonce as a request parameter we could submit a PR to have the PyLTI1p3 optionally use it. I just haven't had the time to work out a fix for that to test and submit to the author. Just low priority.